CRM/Core/EntityTokens.php

```diff
@@ -176,78 +198,56 @@ protected function getFieldMetadata(): array {
     }
-    return $this->fieldMetadata;
+    if ($field) {
+      return $this->fieldMetadata[$field] ?? NULL;
+    }
+    else {
+      return $this->fieldMetadata;
+    }
   }
 
   /**
    * Get pseudoTokens - it tokens that reflect the name or label of a pseudoconstant.
    *
-   * @internal - this function will likely be made protected soon.
-   *
+   * @internal - this function is a bridge for legacy CRM_Utils_Token callers. It should be removed.
+   * @deprecated
    * @return array
    */
   public function getPseudoTokens(): array {
-    $return = [];
-    foreach (array_keys($this->getBasicTokens()) as $fieldName) {
-      if ($this->isAddPseudoTokens($fieldName)) {
-        $return[$fieldName . ':label'] = $this->fieldMetadata[$fieldName]['input_attrs']['label'];
-        $return[$fieldName . ':name'] = ts('Machine name') . ': ' . $this->fieldMetadata[$fieldName]['input_attrs']['label'];
+    $labels = $this->getAllTokens();
+    // Simpler, but doesn't currently pass: $labels = $this->tokenNames;
+    $r = [];
+    foreach ($this->getApiTokens() as $key) {
+      if (strpos($key, ':') !== FALSE || strpos($key, '.') !== FALSE) {
+        $r[$key] = $labels[$key];
       }
     }
-    return $return;
-  }
-
-  /**
-   * Is this a field we should add pseudo-tokens to?
-   *
-   * Pseudo-tokens allow access to name and label fields - e.g
-   *
-   * {contribution.contribution_status_id:name} might resolve to 'Completed'
-   *
-   * @param string $fieldName
-   */
-  public function isAddPseudoTokens($fieldName): bool {
-    if ($fieldName === 'currency') {
-      // 'currency' is manually added to the skip list as an anomaly.
-      // name & label aren't that suitable for 'currency' (symbol, which
-      // possibly maps to 'abbr' would be) and we can't gather that
-      // from the metadata as yet.
-      return FALSE;
-    }
-    return (bool) $this->getFieldMetadata()[$fieldName]['options'];
+    return $r;
   }
 
   /**
-   * Get the value for the relevant pseudo field.
-   *
-   * @param string $realField e.g contribution_status_id
-   * @param string $pseudoKey e.g name
-   * @param int|string $fieldValue e.g 1
+   * Get the values for all exported pseudo-fields.
    *
-   * @return string
-   *   Eg. 'Completed' in the example above.
-   *
-   * @internal function will likely be protected soon.
+   * @param int $id
+   * @return array
+   * @throws \API_Exception
+   * @internal - this function is a bridge for legacy CRM_Utils_Token callers. It should be removed.
+   * @deprecated
    */
-  public function getPseudoValue(string $realField, string $pseudoKey, $fieldValue): string {
-    if ($pseudoKey === 'name') {
-      $fieldValue = (string) CRM_Core_PseudoConstant::getName($this->getBAOName(), $realField, $fieldValue);
-    }
-    if ($pseudoKey === 'label') {
-      $fieldValue = (string) CRM_Core_PseudoConstant::getLabel($this->getBAOName(), $realField, $fieldValue);
+  public function getPseudoValues(int $id): array {
```

Comment:
It looks like this is only called from a different class now - so it should be moved to that class.

Reply:
note there is no reason to deprecate - it's only called from recent tested core code